Fhi.HelseId.Web/Services/HelseIdSecretHandlers.cs

```diff
@@ -23,14 +23,33 @@ public interface IHelseIdSecretHandler
 
     public abstract class SecretHandlerBase : IHelseIdSecretHandler
     {
-        protected JsonWebKey? jwkSecurityKey;
-        protected IHelseIdWebKonfigurasjon? configAuth;
+        protected JsonWebKey _jwkSecurityKey = null!; // Explicitly checks for null when GenerateClientAssertion is called.
+        protected IHelseIdWebKonfigurasjon _configAuth = null!; // Explicitly checks for null when GenerateClientAssertion is called.
 
         public const string ClientAssertionType = IdentityModel.OidcConstants.ClientAssertionTypes.JwtBearer;
 
         public virtual void AddSecretConfiguration(IHelseIdWebKonfigurasjon configAuth, OpenIdConnectOptions options) { }
 
-        public virtual string GenerateClientAssertion => ClientAssertion.Generate(configAuth.ClientId, configAuth.Authority, jwkSecurityKey);
+        public virtual string GenerateClientAssertion
+        {
+            get
+            {
+                if (_configAuth == null)
+                {
+                    throw new ArgumentNullException(nameof(_configAuth) + " cannot be null.");
```

Comment:
The error message her would look like:
```txt
Value cannot be null. (Parameter '_configAuth cannot be null.')
```

The first parameter is the name of the variable that is checked.

So concatenating the name and the message is not quite correct.

If you change to 

```cs
throw new ArgumentNullException(nameof(_configAuth));
```

the message would be:

```txt
Value cannot be null. (Parameter '_configAuth')
```

which should be good enough, so the message there is actually redundant.

Reply:
Beautiful! Removed redundant error messages.